src/core/applications/services/categoriaService.ts

```diff
@@ -0,0 +1,27 @@
+import Categoria from "core/domain/categorias";
+import CategoriaRepository from "../ports/categoriaRepository";
+
+export default class CategoriaService {
+    constructor(private readonly categoriaRepository: CategoriaRepository) { }
+
+    async criaCategoria(categoria: Categoria): Promise<Categoria> {
+        return this.categoriaRepository.criaCategoria(categoria);
+    }
+
+    async deletaCategoria(idCategoria: string): Promise<number> {
+        return this.categoriaRepository.deletaCategoria(idCategoria);
+    }
+
+    async editaCategoria(idCategoria: string, categoria: Categoria): Promise<Categoria | undefined> {
+        return this.categoriaRepository.editaCategoria(idCategoria, categoria);
+    }
```

Comment:
Adicionando a validação do DTO a gente pode mdar os tipos das funções para os respectivos DTOs.

```suggestion
    async editaCategoria(idCategoria: string, categoria: EditaCategoriaDTO): Promise<Categoria | undefined> {
        return this.categoriaRepository.editaCategoria(idCategoria, categoria);
    }
```

Reply:
Acho que faz sentido fazer isso em um outro PR.

Minha dúvida é se isso seria uma validação no ```core/``` ou se seria apenas a validação do body do express e aí ficaria apenas na camada da ```api/```

Reply:
Acho que faz sentido estar na `api/`, assim um `body` não validado nem passaria pras camadas mais internas da app

Reply:
fica como nova tarefa